scripts/src/qc_table_functions.R

```diff
@@ -0,0 +1,374 @@
+#' @param group_cols A character vector specifying the grouping columns (default: `c("pcr_plate", "pcr_well")`).
+#' @param metric A string indicating the column name of the metric to use for calculations (default: `"n"`).
+#' @param cell_line_cols A character vector specifying the column names that define a unique cell line (default: `c("depmap_id", "pool_id")`).
+#' @param count_threshold An integer specifying the minimum count threshold for a cell line to be considered recovered (default: `40`).
+#'
+#' @return A data frame summarizing read statistics for each group, including total reads, expected reads, control barcode reads,
+#' recovered cell lines, and their fractions.
+#'
+#' @import dplyr
+compute_read_stats <- function(annotated_counts, cell_set_meta, group_cols = c("pcr_plate","pcr_well"),
+                               metric = "n", cell_line_cols = c("depmap_id", "pool_id"), count_threshold = 40) {
+  # Compute expected lines from cell_set_meta
+  expected_lines <- compute_expected_lines(cell_set_meta, cell_line_cols)
+
+  result <- annotated_counts %>%
+    dplyr::left_join(expected_lines, by = "cell_set") %>% # Add n_expected_lines from lookup
+    dplyr::group_by(across(all_of(group_cols))) %>%
+    dplyr::summarise(
+      # Total reads
+      n_total_reads = sum(.data[[metric]], na.rm = TRUE),
+      # Reads mapping to cell lines
+      n_bc_reads = sum(.data[[metric]][expected_read], na.rm = TRUE),
+      # Fraction of reads mapping to cell lines
+      fraction_expected_reads = n_bc_reads / n_total_reads,
+      # Reads mapping to control barcodes
+      n_cb_reads = sum(!is.na(cb_name) & cb_name != "", na.rm = TRUE),
+      # Number of cell lines with coverage above 40 reads
+      n_lines_recovered = sum(.data[[metric]] >= count_threshold & (is.na(cb_name) | cb_name == ""), na.rm = TRUE),
+      # Number of expected lines based on metadata
+      n_expected_lines = max(n_expected_lines, na.rm = TRUE), # Bring forward from join
+      # Fraction of cell lines with coverage above count threshold
+      fraction_cl_recovered = n_lines_recovered / max(n_expected_lines, na.rm = TRUE),
+    ) %>%
+    dplyr::ungroup()
+}
+
+#' Extract CB metrics
+#'
+#' This function extracts precomputed control barcode metrics from the normalized counts table.
+#'
+#' @param normalized_counts A data frame containing normalized read counts and associated metrics.
+#' @param group_cols A character vector specifying the grouping columns (default: `c("pcr_plate", "pcr_well")`).
+#' @param metrics A character vector specifying the metric column names to include in the output
+#'   (default: `c("cb_intercept", "norm_mae", "norm_r2", "cb_spearman")`).
+#'
+#' @return A data frame containing unique combinations of `group_cols` and the specified `metrics`.
+#'
+#' @import dplyr
+get_cb_metrics <- function(normalized_counts, group_cols = c("pcr_plate", "pcr_well"), metrics = c("cb_intercept", "norm_mae", "norm_r2", "cb_spearman")) {
```

Comment:
What do you think about moving the computation of QC metrics such as norm_mae, norm_r2, and cb_spearman to this module?
This would mean that we keep only cb_intercept in normalized counts file and read cb_intercept from normalized counts to compute the three QC metrics. 
Feel like this organization will be much clearer.